### PR TITLE
openra: add release- prefix to version string

### PR DESCRIPTION
**Pull request by mikatammi**
Change version string to have prefix `release-`. This fixes "Server is running an incompatible version.", when trying to play against official builds of the OpenRA.

## Description of changes

openra: Add release- prefix to version string

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`): Tested basic multiplayer functionality against OpenRA running on macOS
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mikatammi** [comment]
Note: I tested this on top of the 20231010 version

**Review by mdarocha — CHANGES_REQUESTED** [review]

File: pkgs/games/openra/build-engine.nix [hunk 1]
```diff
@@ -36,7 +36,7 @@ buildDotnetModule rec {
   dontDotnetFixup = true;
 
   preBuild = ''
-    make VERSION=${version} version
+    make VERSION=release-${version} version
```
[the remark this answers is not in the document]
Comment (mikatammi) [reply pkgs/games/openra/build-engine.nix:39]: What would be a good way to check it?

[the remark this answers is not in the document]
Comment (mikatammi) [reply pkgs/games/openra/build-engine.nix:39]: Maybe we can just use ${engine.build} ?

[the remark this answers is not in the document]
Comment (mikatammi) [reply pkgs/games/openra/build-engine.nix]: Changed to `${engine.build}-${version}`

**Review by mdarocha — APPROVED** [review]

**Pull request merged**